gmail_fetcher.py

```
@@ -411,9 +403,21 @@ def print_summary(hours: int, stats: dict):
         print(tabulate(table, headers=['Category', 'Count'], tablefmt='grid'))
     print("="*50 + "\n")
 
-def main(email_address: str, app_password: str, hours: int = 2):
+def test_api_connection(api_token: str) -> None:
+    """Test connection to control API before proceeding"""
+    service = DomainService(api_token=api_token)
+    try:
+        # Try to fetch domains to verify API connection
+        service.fetch_allowed_domains()
+    except Exception as e:
+        raise Exception(f"Failed to connect to control API: {str(e)}")
```

Ruff (0.7.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)